### PR TITLE
Remove CrateComponent load mechanism

### PR DESCRIPTION
**Pull request by mfussenegger**
Instead load all componenets via CrateNode. The dependencies are all
known and available at compile time.

There is no need to use a discovery mechanism to find them.

### CODE REVIEW

**Comment by mfussenegger** [comment]
@dobe please review

File: core/build.gradle [hunk 1]
```diff
@@ -62,6 +62,7 @@ task getVersion (dependsOn: 'classes') << {
 
 sourceSets {
     test {
+        output.resourcesDir = null
```
Comment (dobe) [inline core/build.gradle]: if there are no resources you should not specify srcDir

**Comment by mfussenegger** [comment]
@dobe can you take a look at this again?

**Comment by dobe** [comment]
ack

**Pull request merged**